packages/create-experience-builder/templates/react-vite-ts/src/App.tsx

```diff
@@ -28,14 +30,23 @@ function App() {
       : import.meta.env.VITE_ACCESS_TOKEN,
   });
 
-  const { experience, defineComponents } = useExperienceBuilder({
-    experienceTypeId,
-    client,
-    mode,
-  });
+  const [experience, setExperience] = useState<Experience<EntityStore>>();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const experience = await fetchBySlug({
+        client,
+        experienceTypeId,
+        slug: 'homePage',
+        localeCode: 'en-US',
+        mode,
+      });
+
+      setExperience(experience);
+    };
 
-  // Register optional default components
-  useExperienceBuilderComponents(defineComponents);
+    fetchData();
+  }, [client, mode]);
```

Comment:
Should this `create-experience-builder` app use the `useFetchBySlug` hook instead of calling `fetchBySlug` directly? 🤔

Reply:
fo sho, see the wiki set up instructions on how: https://github.com/contentful/experience-builder/wiki#fetch-and-display-the-experience

Reply:
Actually, i have a branch where this is already updated, heres the code you should be able to use as it for `app.tsx`:

```
import { createClient } from 'contentful';
import { useFetchBySlug, ExperienceRoot } from '@contentful/experience-builder';
import './App.css';

// Import the styles for the default components
import '@contentful/experience-builder-components/styles.css';

const experienceTypeId = import.meta.env.VITE_CTFL_EXPERIENCE_TYPE_ID; //Content type id for the experience
const localeCode = 'en-US'; //Locale code for the experience (could be dynamic)

const client = createClient({
  // your space id
  space: import.meta.env.VITE_CTFL_SPACE_ID,
  // your environment id
  environment: import.meta.env.VITE_CTFL_ENV_ID,
  // Supported values: 'preview.contentful.com' or 'cdn.contentful.com',
  host: import.meta.env.VITE_CTFL_API_HOST,
  // needs to be access token if host = 'cdn.contentful.com' and preview token if 'preview.contentful.com'
  accessToken: import.meta.env.VITE_CTFL_ACCESS_TOKEN,
});

function App() {
  const { experience, isLoading, error } = useFetchBySlug({
    client,
    slug: 'homePage', //Could be fetched from the url,
    experienceTypeId,
    localeCode,
  });

  // handle loading and error states
  if (isLoading) return <div>Loading...</div>;
  if (error) return <div>Error: {error.message}</div>;

  return <ExperienceRoot experience={experience} locale={localeCode} />;
}

export default App;

```